### DISCUSSION TITLE
Could we bring back the diamond background of the icon.

### DISCUSSION THREAD

**Discussion by UncleBill** [macvim-dev/macvim#1140] [General]
Hi there!

All designs by @jasonlong are awesome! But I like the [`option 5`][1] the most. It bases on the original one. And the the shining `V` looks vividly on the green diamond. Some area of the letter over the diamond makes a vision of stereo. On the other hand, it is more similar to the original icon, the transition would be smoother to us.

I thinks the votes in [#1054][2] are too few, it's hard to represent the most, So I would like to raise another poll here. To make the compare clear, I put them all together:

| original    | :+1:        | :smile:     | :tada: _(current)_      | :heart:     | :rocket:    |
| ----------- | ----------- | ----------- | ----------- | ----------- | ----------- |
| ![image][8] | ![image][3] | ![image][4] | ![iamge][5] | ![image][6] | ![image][7] |

**(Vote for an icon by choose the emoji above)**

[1]: https://github.com/macvim-dev/macvim/pull/1054#issuecomment-703217863
[2]: https://github.com/macvim-dev/macvim/pull/1054
[3]: https://user-images.githubusercontent.com/6104/86523031-5d3bd300-be34-11ea-92eb-62087bd43a9c.png
[4]: https://user-images.githubusercontent.com/6104/86540712-6e88ec00-bed5-11ea-9521-d454a89ca975.png
[5]: https://user-images.githubusercontent.com/6104/86777331-0518f280-c027-11ea-9552-e9c137f1ed99.png
[6]: https://user-images.githubusercontent.com/6104/87156741-237f2800-c28b-11ea-92f2-94b15344fe3e.png
[7]: https://user-images.githubusercontent.com/6104/87954691-b59ce280-ca7a-11ea-9031-d29009ef396d.png
[8]: https://user-images.githubusercontent.com/1141198/101988041-5f9f3480-3cd2-11eb-86a7-7d08c67007c5.png

**Comment by ychin** [comment]
Nice use of the discussion feature! :) Yeah this is kind of subjective and hard to get enough people to vote on these kinds of things. I'll probably let it sit a bit and see what people vote on.

**Reply by ychin** [reply]
Also, I looked it up and I don't believe Apple allows you to specify alternate icons. Confusingly, that's possible for iOS apps (e.g. Pandora has a set of icons that you can choose from), but for macOS apps you have to pick one. You could change the icon during runtime but it wouldn't persist in the Dock after you close the app.

You *could* customize the app's icon yourself, if you go to go to the "Get Info", and manually drag an icon over, and it will persist that way, but then it won't be the default icon being used. See https://9to5mac.com/2019/01/17/change-mac-icons/

**Comment by gouch** [comment]
The diamond Vim logo worked as an icon in the past _because_ of the transparent background. In the contexts of the Dock, Launchpad or Spotlight, the color behind the icon is rarely #fff, so there's enough contrast that you can make out the parts of the V that extend outside the diamond.

Adding a white square behind the diamond creates a permanent low contrast context, which makes the corners of the light gray V much harder to perceive, especially as the icon gets smaller. (For comparison, see how the [Vim logo](https://commons.wikimedia.org/wiki/File:Vimlogo.svg) uses both a darker gray and has a solid black outline as well, which helps its contrast against white.) To me, the diamond-on-white options need to be careful (with color, stroke, or shadow work) to make sure the light gray V corners aren't lost against white.

(If I got to make a second, looser argument🙂,  I'd say the green square options feel like they're designed with Big Sur's new icon constraint in mind and make good use of the available canvas.)

All that being said, all of @jasonlong's work and effort is awesome to see and a wonderful gift!

**Comment by diogocampos** [comment]
As I've already mentioned [here](https://github.com/macvim-dev/macvim/pull/1054#issuecomment-743927088), I think a version with parts of the _V_ sticking out of the square could be interesting and reminiscent of the previous icon.

Think outside the square :)

**Reply by ychin** [reply]
I think the issue I see with that is that most Big Sur icons don't do that (while the HIG says it's ok, it's mostly for limited cases), and I'm not sure if there is a way to get the _V_ to poke out without having to align it off center or change the typography (which is what identifies the "V" as "Vim").

**Comment by jsit** [comment]
I think the "diamond" is such a part of Vim's brand legacy that it feels like something is lost not to include it. I think ["Option ❤️"](https://user-images.githubusercontent.com/6104/87156741-237f2800-c28b-11ea-92f2-94b15344fe3e.png) really ought to have more consideration.
